Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/Curvatures/interpolated_corrected_curvature_measures.h

```diff
@@ -0,0 +1,973 @@
+*
+* @tparam GT is the geometric traits class.
+*
+* @param x is a vector of the vertex positions of the face.
+* @param u is a vector of the vertex nomrals of the face.
+*
+* @return a scalar of type `GT::FT`.
+* This is the value of the interpolated corrected area measure of the given face.
+*
+* @see `interpolated_corrected_mean_curvature_measure_face()`
+* @see `interpolated_corrected_gaussian_curvature_measure_face()`
+* @see `interpolated_corrected_measure_mesh()`
+*/
+template<typename GT>
+typename GT::FT interpolated_corrected_area_measure_face(const std::vector<typename GT::Vector_3>& u,
+                                                         const std::vector<typename GT::Vector_3>& x = {})
+{
+    std::size_t n = x.size();
+    CGAL_precondition(u.size() == n);
+    CGAL_precondition(n >= 3);
+
+    typename GT::Construct_cross_product_vector_3 cross_product;
+
+    // Triangle: use triangle formula
+    if (n == 3)
+    {
+        const typename GT::Vector_3 um = (u[0] + u[1] + u[2]) / 3.0;
+        return 0.5 * um * cross_product(x[1] - x[0], x[2] - x[0]);
+    }
+    // Quad: use bilinear interpolation formula
+    else if (n == 4)
+    {
+        // for the formulas below, values of verices 2 & 3 are swapped (compared to paper) to correct order.
+        // the indices in paper vs in here are: 00 = 0, 10 = 1, 11 = 2, 01 = 3
+        return (1.0 / 36.0) * (
+              (4 * u[0] + 2 * u[1] + 2 * u[3] + u[2]) * cross_product(u[1] - u[0], u[3] - u[0])
+            + (2 * u[0] + 4 * u[1] + u[3] + 2 * u[2]) * cross_product(u[1] - u[0], u[2] - u[1])
+            + (2 * u[0] + u[1] + 4 * u[3] + 2 * u[2]) * cross_product(u[2] - u[3], u[3] - u[0])
+            + (u[0] + 2 * u[1] + 2 * u[3] + 4 * u[2]) * cross_product(u[2] - u[3], u[2] - u[1])
+            );
+    }
+    // N-gon: split into n triangles by polygon center and use triangle formula for each
+    else
+    {
+        typename GT::FT mu0 = 0;
+
+        // getting center of points
+        typename GT::Vector_3 xc =
+            std::accumulate(x.begin(), x.end(), typename GT::Vector_3(0, 0, 0));
+        xc /= n;
+
+        // getting unit average normal of points
+        typename GT::Vector_3 uc =
+            std::accumulate(u.begin(), u.end(), typename GT::Vector_3(0, 0, 0));
+        uc /= sqrt(uc * uc);
+
+        // summing each triangle's measure after triangulation by barycenter split.
+        for (std::size_t i = 0; i < n; i++)
+        {
+            mu0 += interpolated_corrected_area_measure_face<GT>(
+                std::vector<typename GT::Vector_3> {u[i], u[i + 1 % n], uc},
+                std::vector<typename GT::Vector_3> {x[i], x[i + 1 % n], xc}
+            );
+        }
+        return mu0;
+    }
+}
+
+/**
+* \ingroup PMP_corrected_curvatures_grp
+*
+* computes the interpolated corrected mean curvature measure (mu1) of a specific face.
+*
+* @tparam GT is the geometric traits class.
```

Comment:
```suggestion
* @tparam GT geometric traits class model of `Kernel`
```

Reply:
It is a bit overkill and you can define a concept with only the things you are using.